### DISCUSSION TITLE
ResumePausedMutations with depedencies

### DISCUSSION THREAD

**Discussion by woen4** [TanStack/query#4012] [General]
I need to support my app users offline. For that I'm using react query with offlineFirst mode. But I need mutations fired when offline to run before my queries on reconnect (I'm using refetchOnReconnect)

> ### Mutations Run -> after this -> RefetchOnReonnect run. 

What we can made to allow this behaviour?

**Comment by TkDodo** [comment]
not sure if this is easily possible at the moment. Might be best to not refetchOnReconnect if you have paused mutations, as you'll likely refetch after the successful mutation anyways (in onSuccess or onSettled). Worst case, it would refetch (on reconnect), then run mutations, then refetch again ?
